### ISSUE TITLE
python requests and HTTPRequest libraries fails when used with moto, mock_s3

### ISSUE THREAD

**Issue by jagadish432** [spulec/moto#3530]
## Reporting Bugs

python requests and HTTPRequest libraries fails when used with moto, mock_s3.

1. when used "@mock_s3" directive and when I run the code, the existing code to fetch data from URL using python requests library is failing with the error - "TypeError: cannot unpack non-iterable CallbackResponse object"
2. when I comment this "@mock_s3" directive, then everything is working as normal and expected.

  - I am using the moto with version - 1.3.16 and I installed it using "pip install moto" in windows.

**Comment by bblommers** [comment]
Hi @jagadish432, I believe this was fixed with #3466. I can't reproduce this anymore using moto >= 1.3.16.dev122. 

Are you able to upgrade to that version and verify that it fixes your problem?

**Comment by arokem** [comment]
We seem to still be hitting this (using `mock_ecr`, [here](https://github.com/nrdg/cloudknot/pull/268) -- although the code in the PR is mostly irrelevant to this issue), even after upgrading to that version. Thanks!

**Comment by bblommers** [comment]
Hi @arekom, did some debugging with the test cases in your repo (on a Linux machine). 
Running `make test` with the latest moto installed gives me these errors:
```
FAILED cloudknot/tests/test_config.py::test_is_valid_stack - botocore.exceptions.ClientError: An error occurred (EntityAlreadyExists) when calling the DeleteStack operation: Role with name test-valid-stack-ecs-instance-role already exists.
FAILED cloudknot/tests/test_config.py::test_prune_stacks - botocore.exceptions.ClientError: [.. same ..]
FAILED cloudknot/tests/test_knot.py::test_knot - botocore.exceptions.ClientError: [.. same ..]
FAILED cloudknot/tests/test_pars.py::test_pars_with_default_vpc - botocore.exceptions.ClientError: [.. same ..]
FAILED cloudknot/tests/test_pars.py::test_pars_with_new_vpc - botocore.exceptions.ClientError: [.. same ..]
```
Looks like this is due to the fact that moto does not support deletion of IAM InstanceProfiles/Roles. I've raised #3583 to reflect this.
Your tests do succeed for me after monkeypatching support for that in Moto.

Were you getting the same error, `TypeError: cannot unpack non-iterable CallbackResponse object`? On all tests, or only specific ones?

**Comment by arokem** [comment]
Hi @bblommers. Thanks for looking into this! 

Yes - I can confirm that if I pin responses==0.12.0 and upgrade to moto==1.3.16 I see just the errors you mentioned (and not the `TypeError: cannot unpack non-iterable CallbackResponse object`)

However, I think that this is a separate issue from the one raised originally. When I then upgrade to responses==0.12.1 I see the errors you mentioned, but also the following: 

```
FAILED cloudknot/tests/test_docker_image.py::test_DockerImage - docker.errors.DockerException: Error while fetching server API version: cannot unpack non-iterable CallbackResponse object
FAILED cloudknot/tests/test_knot.py::test_knot - TypeError: cannot unpack non-iterable CallbackResponse object
```

So, I think there might actually be two separate issues here (?)

**Comment by bblommers** [comment]
Do you have Docker running? I.e., does `docker info` pass? 
The first exception looks like a bug in moto where it doesn't handle the failure if it can't communicate with the docker socket.

On the second exception:
> If I [..] upgrade to moto==1.3.16
Just to make it more explicit: our version naming is slightly off. `1.3.16.dev122` is a later release then `1.3.16`. So 1.3.16 would give this exception with the latest responses, but 1.3.16.dev122 contains the fix for that.

**Comment by arokem** [comment]
OK - when I upgrade to 1.3.16.dev122 I see only the same errors that you see, so I think that:

1. It looks like the issue in the OP here is resolved in 1.3.16.dev122; and 
2. Once #3583 is resolved, we should be all set on our end. 

So, I think that you can go ahead and close this issue, and then we can follow up on #3583 once that's resolved. 

Thanks so much for your work on this!
